src/main/java/hudson/tasks/test/AbstractTestResultAction.java

```diff
@@ -509,4 +1593,46 @@ public Object readResolve() {
         }
     }
 
+    /**
+     * A generic class to encapsulate a pair of objects where each object can be any type.
+     * @param <A> Type of the first object.
+     * @param <B> Type of the second object.
+     */
+    private static final class Pair<A,B>{
```

Comment:
Why create another Pair, is there no similar one in classpath ?

Reply:
To use any predefined Pair data structure some new package needs to be included so, I made a small class for it.